File: Espoir/Entities/NonPlayablePlayer.cs
namespace Espoir.Entities
{
    internal class NonPlayablePlayer : Player
    {
        public static NonPlayablePlayer Generate(int id)
        {
            var warFunds = new Decimal((Random.Shared.Next() % 9000001) + 1000000);
            return new NonPlayablePlayer()
            {
                Name = $"NPC#{id}",
                Cards = CardHelper.GenerateStartingHand(),
                WarFunds = warFunds,
                Debt = warFunds,
            };
        }

        public CardType GetRandomCard()
        {
            return this.Cards[Random.Shared.Next() % this.Cards.Count];
        }
    }
}

File: renpy-version/entities/non_playable_player.py
from random import randrange
from entities.player import Player
from helpers.card_helper import generateStartingHand

class NonPlayablePlayer(Player):
    def __init__(self, id):
        super().__init__()
        
        warfunds = randrange(1e6, 1e7+1)
        self.Name = f'NPC#{id}'
        self.Cards = generateStartingHand()
        self.WarFunds = warfunds
        self.Debt = warfunds
    
    def getRandomCard(self):
        return self.Cards[randrange(len(self.Cards))]